From da112737eb109dc263169e820b7ef903b7f8624f Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter@cs.unisa.edu.au>
Date: Sun, 8 Jun 2008 00:12:03 +0930
Subject: [PATCH] Check for XINPUT ABI < 3 (corrected version)

Thanks to Sven Wegener for pointing out the incorrect previous version.
---
 src/evdev.c |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/evdev.c b/src/evdev.c
index 13f03d3..b146d41 100644
--- a/src/evdev.c
+++ b/src/evdev.c
@@ -726,7 +726,10 @@ EvdevAddAbsClass(DeviceIntPtr device)
     pEvdev->min_y = absinfo_y.minimum;
     pEvdev->max_y = absinfo_y.maximum;
 
-    if (!InitValuatorClassDeviceStruct(device, 2, GetMotionHistory,
+    if (!InitValuatorClassDeviceStruct(device, 2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+                                       GetMotionHistory,
+#endif
                                        GetMotionHistorySize(), Absolute))
         return !Success;
 
@@ -756,7 +759,10 @@ EvdevAddRelClass(DeviceIntPtr device)
 
     pInfo = device->public.devicePrivate;
 
-    if (!InitValuatorClassDeviceStruct(device, 2, GetMotionHistory,
+    if (!InitValuatorClassDeviceStruct(device, 2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+                                       GetMotionHistory,
+#endif
                                        GetMotionHistorySize(), Relative))
         return !Success;
 
-- 
1.5.5.1

